apps/api/internal/site/monitoring_guards_test.go

```
@@ -0,0 +1,497 @@
+	engine := gin.New()
+	engine.Use(func(c *gin.Context) {
+		c.Request = c.Request.WithContext(domain.WithPrincipal(c.Request.Context(), p))
+		c.Next()
+	})
+	engine.POST("/pause", h.pauseMonitoring)
+	engine.POST("/resume", h.resumeMonitoring)
+
+	for _, path := range []string{"/pause", "/resume"} {
+		body, _ := json.Marshal(map[string]any{"site_ids": []string{siteID.String()}})
+		rec := httptest.NewRecorder()
+		req := httptest.NewRequest(http.MethodPost, path, bytes.NewReader(body))
```

golangci-lint (2.12.2)

[error] net/http/httptest.NewRequest must not be called. use net/http/httptest.NewRequestWithContext

(noctx)